### ISSUE TITLE
Photo Losing Quality When Setting As Group Picture

### ISSUE THREAD

**Issue by DeMoN100** [simplex-chat/simplex-chat#2118]
![Output1680312064](https://user-images.githubusercontent.com/8564572/229258938-67a9dc1c-c15b-476d-95a7-9a30777d700f.gif)
Not sure if this happens with only .png photos but I think this happens with jpg as well.

**Comment by epoberezkin** [comment]
yes, it's possible to improve, but seems too complex/too much effort for what's it worth.

The images in the profiles are compressed to fit in a single 16kb message, together with lots of other stuff, and base64 encoded. You can certainly achieve a much better quality by compressing it manually - target size for binary JPEG is ~9300 bytes (or exactly 12500 bytes as base64 image - you can test it in the browser). But even if you compress it to 20-30kb and it looks ok, it's likely to look ok in the app too.

**Comment by DeMoN100** [comment]
> yes, it's possible to improve, but seems too complex/too much effort for what's it worth.

Backburner it on the roadmap. One more milestone to achieve knowledge shall be gained from this difficulty.

**Comment by epoberezkin** [comment]
That's a relatively small thing, but we can indeed add "high[er]-res profile pictures" to the roadmap :)

**Comment by epoberezkin** [comment]
btw, PNGs are particularly bad, because of the lack of lossy compression, using JPG usually results in a better visual quality.

**Comment by DeMoN100** [comment]
> btw, PNGs are particularly bad, because of the lack of lossy compression, using JPG usually results in a better visual quality.

Right because you programmed the jpg already which I saw you set to 85% quality almost unnoticed loss of detail but I like to be precise. Jpg does not support transparency so switching to a jpg instead is not the answer if I want a png. if there was ever an art group people would want to attach & just send  rather then have to use a third party just to not lose quality in the photo.

**Comment by atlasveldine** [comment]
I get that this is a seen as a pretty low importance issue and, given that I don't personally have the necessary skill required to muck about with Haskell, and it seems that the devs here don't believe this to be important, I don't expect that it's going to be resolved any time soon. Still, I personally find this problem to be incredibly irritating, and am hopeful that maybe speaking up might bring this issue to the forefront and help get it resolved faster. Profile pictures consistently become so heavily compressed that they're almost always a blurry mess, and it really detracts from the UX whenever interacting with the UI elements that contain the larger profile pictures (so, contact views, conversation list view, and profile picture change views).

This doesn't seem to be limited to group photos, either, so perhaps this issue's title should be renamed. Anytime I try to change one of my profile's or groups' profile pictures I inevitably spend way too much time trying to optimize it to reduce the result's heavy, blurry compression. 

It's particularly offensive on desktop and on high resolution/large screen size mobile devices, like tablets. It's 2024, there's really no reason why any image, let alone one which is sent over the Internet very infrequently, should be locked to such an absurdly tiny size resolution. 

Every single contact that I have on SimpleX Chat has an ugly, blurry mess for their profile pictures. 

What can be done to improve or resolve this problem? What is the associated difficulty involved? Is there a genuinely good reason why this issue hasn't been resolved and why it's seen as low importance? The way I see it, these sorts of weird tidbits are issues that will subtly turn people off of using the platform. A new user who spends fifteen minutes trying to get their profile picture to not be an ugly mess may well uninstall the app and give up on it for that reason alone. It's unnecessary friction, and in the modern era people tend to have extremely low patience for these kinds of things — they're very quick to judge something negatively over what might seem like tiny problems to developers. It's certainly not a deal breaker for myself, but for, say, Aunt Mary with little technological knowledge, being unable to use a photo of her face that doesn't become a horrifying blur might be more than enough reason to drop the app.

It seems to me that this should be a relatively simple thing to resolve, though I will readily acknowledge that I don't fully grasp the underlying protocol. If I understand right, then at present it's being sent in a max 16kb message, which is why it's being compressed so horribly. Changing this outright would presumably break compatibility with older versions of the protocol, so that's not an ideal option. What about instead extending the protocol? Continue to send the horribly compressed version as it has been, but following that, send a new message type following a successful connection utilizing XFTP with the full image (or a larger, but still compressed, image, if there's some concern over the data usage overhead of the servers). This way, older clients will continue functioning, but newer clients will be able to overwrite the image data for that contact with the improved version.
